docs/source/development/poetry.rst

```diff
@@ -0,0 +1,108 @@
+
+We have a :ref:`Poetry style guide<poetry-shell-link>` for specific recommendations about using Poetry in these projects.
+
+.. _using-poetry-link:
+
+Using Poetry
+------------
+
+Poetry can be installed with the `official instructions <https://python-poetry.org/docs/master/>`_.
+
+The dependencies for the project are all specified in the ``pyproject.toml`` file in the base level of the repository.
+
+The Poetry project has a great `basic usage guide <https://python-poetry.org/docs/master/basic-usage/>`_ which introduces the basic capabilities. This covers the basic commands which are used in installing and managing Poetry.
+
+You can use Poetry in addition to other Python tools. Unlike with Conda, installing Poetry should not interfere with other projects which may use other tools. You can also use different methods for virtual environments if you prefer, which is covered under the Poetry shell section. Poetry uses pip to install under the hood.
+
+.. _poetry-dependency-link:
+
+Adding Dependencies
+-------------------
+
+Poetry has a command to `add a new dependency <https://python-poetry.org/docs/master/cli/#add>`_, with an optional version specification:
+
+::
+
+    # Add a new dependency
+    poetry add pendulum@^2.0.5
+
+    # Add a new dependency with default latest version
+    poetry add pendulum
+
+These dependencies are then added automatically to the ``pyproject.toml`` file. The overall project dependencies go under ``[tool.poetry.dependencies]``. The main project dependencies are always installed.
+
+You can also update the ``pyproject.toml`` file directly, using the existing formatting or the `Poetry documentation on it <https://python-poetry.org/docs/pyproject/>`_ as a guide.
+
+After you update any dependencies, you will need to update the lock file::
+    poetry lock
+
+This will create a new version of the ``poetry.lock`` file, which should be committed to the repository. Our pre-commit tools also do this step automatically if needed.
+
+.. _poetry-dependency-groups-link:
+
+Dependency groups
+^^^^^^^^^^^^^^^^^^
+
+Poetry also provides dependency groups for seperating dependencies into logical seperations. If you are installing the project as an end user, you do not need the development tools. The testing environment does not need the documentation generation dependencies. In our case, the AWS Lambda environment does not need the same dependencies as the CDK deployment. Before you add a dependency to the main group, ask yourself if it would make more sense in one of the other existing dependency groups.
```

Comment:
```suggestion
Poetry also provides dependency groups for separating dependencies into logical separations. If you are installing the project as an end user, you do not need the development tools. The testing environment does not need the documentation generation dependencies. In our case, the AWS Lambda environment does not need the same dependencies as the CDK deployment. Before you add a dependency to the main group, ask yourself if it would make more sense in one of the other existing dependency groups.
```

Reply:
"separate" gets me every time! Thanks